### PR TITLE
Lex toposort

### PR DESCRIPTION
**Pull request by vshampor**

### CODE REVIEW

**Comment by vshampor** [comment]
Jenkins please stop all builds

**Comment by vshampor** [comment]
Jenkins please stop all jobs

**Comment by vshampor** [comment]
Jenkins please retry a build

File: tests/pruning/test_model_pruning_analysis.py [hunk 1]
```diff
@@ -108,8 +108,10 @@ def __init__(self, model, not_pruned_nodes, pruned_groups, pruned_groups_by_node
                                   not_pruned_nodes=[],
                                   pruned_groups=[['1 TestModelEltwiseCombination/NNCFConv2d[conv1]/conv2d_0',
                                                   '2 TestModelEltwiseCombination/NNCFConv2d[conv2]/conv2d_0',
-                                                  '4 TestModelEltwiseCombination/NNCFConv2d[conv3]/conv2d_0',
-                                                  '6 TestModelEltwiseCombination/NNCFConv2d[conv4]/conv2d_0'],
+                                                  # Order is determined by runtime cluster merge logic.
+                                                  # TODO: should the pruned module info be delivered sorted?
+                                                  '6 TestModelEltwiseCombination/NNCFConv2d[conv4]/conv2d_0',
+                                                  '4 TestModelEltwiseCombination/NNCFConv2d[conv3]/conv2d_0'],
```
Comment (vshampor) [inline tests/pruning/test_model_pruning_analysis.py:114]: @evgeniya-egupova @mkaglins FYI

**Comment by vshampor** [comment]
Jenkins please retry a build

**Comment by vshampor** [comment]
Jenkins please stop all jobs

**Comment by vshampor** [comment]
Jenkins please stop all builds

**Comment by FyodorK** [comment]
Jenkins please stop all builds

**Comment by vshampor** [comment]
Jenkins please retry a build

**Pull request merged**